task/upload.go

```diff
@@ -76,6 +76,14 @@ func handleUploadVOD(p handleUploadVODParams) (*TaskHandlerOutput, error) {
 		if err != nil {
 			return nil, err
 		}
+		var encryption *clients.EncryptionPayload
+		params := *tctx.Task.Params.Upload
+
+		if params.Encryption.EncryptedKey != "" {
+			encryption = &clients.EncryptionPayload{
+				EncryptedKey: params.Encryption.EncryptedKey,
```

Comment:
You need to check whether the upload params are not nil. I've made that mistake with that old experiment and it would break transcode API (should we support encryption in transcode API?)

Also while at it, do you mind finding where the old Encryption.Key was used and possibly deleting that code? Or can be done as a follow-up (there are other repos to update)

Reply:
Of course! Yeah, the old code needs to be deleted both here and on studio, currently on studio it's just commented out on the experiment check